### ISSUE TITLE
Test #1 regression fails on architectures mipsel, i386, armel and armhf

### ISSUE THREAD

**Issue by babelouest** [libical/libical#606]
Hello,

While packaging libical 3.0.15 for Debian, I encountered several architecture build fail due to test regression returning an error [[1]](https://buildd.debian.org/status/package.php?p=libical3)

The log output tells the test fail [[2]](https://buildd.debian.org/status/fetch.php?pkg=libical3&arch=armel&ver=3.0.15-1&stamp=1665331979&raw=0) like this:

```
########## Test icaltime_as_timet                   (2) ##########
not ok 6 - icaltime_from_string translates 19020101T000000Z to -2145916800
# test failed: "icaltime_as_timet(icaltime_from_string("19020101T000000Z")) == -2145916800"
#          at: ./src/test/regression.c:4963
not ok 7 - icaltime_from_string translates 19290519T000000Z to -1281916800
# test failed: "icaltime_as_timet(icaltime_from_string("19290519T000000Z")) == -1281916800"
#          at: ./src/test/regression.c:4964
not ok 8 - icaltime_from_string translates 19561004T000000Z to -417916800
# test failed: "icaltime_as_timet(icaltime_from_string("19561004T000000Z")) == -417916800"
#          at: ./src/test/regression.c:4965
ok 9 - icaltime_from_string translates 19700101T000000Z to 0
ok 10 - icaltime_from_string translates 19700301T235959Z to 5183999
ok 11 - icaltime_from_string translates 19970717T235959Z to 869183999
ok 12 - icaltime_from_string translates 20241202T235959Z to 1733183999
ok 13 - icaltime_from_string translates 20371231T235959Z to 2145916799

########## Test time                                (3) ##########
```

The last release 3.0.14 tests was running fine in all arch though.

Is there a way to fix this?

Thanks in advance!

[1] - https://buildd.debian.org/status/package.php?p=libical3
[2] - https://buildd.debian.org/status/fetch.php?pkg=libical3&arch=armel&ver=3.0.15-1&stamp=1665331979&raw=0

**Comment by minichma** [comment]
The failing tests are new in 3.0.15. They seem to fail on architectures with 32-bit time_t. I'll check tomorrow to see, whether it's a problem with the tests, a problem in new code or an problem in old code being detected by the new tests.

**Comment by minichma** [comment]
Analyzed the issue and found, that it's a problem of the new tests. On architectures having a 32-bit time_t the function `icaltime_as_timet` only supports dates starting with 1970, while with 64-bit time_t we support dates starting with 1902. The new test doesn't make this distinction an therefore fails. I'll provide a fix probably later today or tomorrow.

> Is there a way to fix this?

If you're asking for a quick workaround, you could conditionally compile [these lines](https://github.com/libical/libical/blob/408c598eea3d40c728df4e7dfd78dcdcc132c377/src/test/regression.c#L4963-L4965) only `#if (SIZEOF_TIME_T > 4)`.

**Comment by minichma** [comment]
@winterz [In icaltime.c we reject dates < 1970 if SIZEOF_TIME_T == 4](https://github.com/libical/libical/blob/408c598eea3d40c728df4e7dfd78dcdcc132c377/src/libical/icaltime.c#L126). Without the guard the function would work just fine starting with 1902, just like in the 64-bit case. So as a fix we could either remove (or update) the guard or update the tests. I'd go for updating the guard to check in the 32-bit case for >= 1902 and < 2038-01-18, and in the 64-bit case (which isn't guarded as of today) for >= 1902 and < year 10000. What do you think?

**Comment by minichma** [comment]
Prepared a fix in https://github.com/libical/libical/pull/607.

**Comment by babelouest** [comment]
The workaround works for 32 bits architectures, thanks @minichma !

**Comment by winterz** [comment]
fixed with #607
